tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -1374,21 +1415,41 @@ def test_throttling(self, cloud_storage_type):
         num_partitions = self.topics[0].partition_count
         topic_name = self.topics[0].name
         rpk.alter_topic_config(topic_name, 'retention.local.target.bytes',
-                               0x1000)
+                               4096)
 
         for pix in range(0, num_partitions):
             wait_for_local_storage_truncate(self.redpanda,
                                             self.topic,
-                                            target_bytes=0x2000,
+                                            target_bytes=8192,
                                             partition_idx=pix,
                                             timeout_sec=30)
 
-        self.logger.info("Start consumer")
+        # A warm-up consume to make all future measurements fairer.
+        # This potentially warms up object storage. Potentially some local
+        # files which are not trimmed yet.
         self.consume()
 
-        times_throttled = self.get_throttling_metric()
-        self.logger.info(f"Consumer was throttled {times_throttled} times")
-        assert times_throttled > 0, f"Expected consumer to be throttled, metric value: {times_throttled}"
+        unrestricted_bw = self.measure_consume_throughput()
+
+        # Limit bandwidth to half of the unrestricted throughput. The expected
+        # throughput depends on the number of CPU cores available and the
+        # number of partitions. In the best case scenario, we expect each
+        # partition to land on a separate core and the throughput limit to
+        # apply to each partition independently.
+        per_shard_bw_limit = int(unrestricted_bw / 2 / num_partitions)
+        expected_bw = per_shard_bw_limit * num_partitions * 1.1
+        self.logger.info(
+            f"Unrestricted bandwidth: {unrestricted_bw} bytes/sec. "
+            f"Configuring per shard limit to {per_shard_bw_limit} bytes/sec. "
+            f"Expecting throughput to be less than {expected_bw} bytes/sec.")
+
+        self.redpanda.set_cluster_config(
+            {'cloud_storage_max_throughput_per_shard': per_shard_bw_limit})
+
+        restricted_bw = self.measure_consume_throughput()
```

Comment:
maybe it makes sense to use more than one data point here just in case? by data point I mean two or three different throughput values

Reply:
it does 2 runs, right? seems reliable enough; hate to waste CI time